index.js

```diff
@@ -122,20 +122,26 @@ class CMSClient {
       const resourceData = await fetch(this.url);
       const headers = await resourceData.headers;
 
-      this.fetched.fields = JSON.parse(
-        headers.get('x-soda2-fields').split(','),
-      );
+      if (headers.get('x-soda2-fields')) {
+        this.fetched.fields = JSON.parse(
+          headers.get('x-soda2-fields').split(','),
+        );
+      }
+
       this.isOutdated = headers.get('x-soda2-data-out-of-date');
       this.lastModified = headers.get('Last-Modified');
 
       await this._fetchResourceMetadata();
 
-      switch (this.type) {
+      switch (this.fetchOptions.output) {
```

Comment:
Since we are just checking a string here, how about using an object lookup that returns an arrow function based on the selected output option (and can default to json as you've done)?

Also, maybe the switch case works better here for now without needing to do any refactoring.

Reply:
Shane - for my learning purposes, can you explain what you mean by 
#returns an arrow function based on the selected output option
? 

Re: switch statements, I'm not opposed to using them here - I don't see any issues with the code - I just know they can cause issues in other situations so some org's decide against using them completely. Not a stickler, for better or for worse.

Reply:
@shanejearley Is this similar to what you mentioned? This is a good refactor but since we want to perform some action based on the type don't we have to do another check after this lookup? 
```js
const resolveOutputType = (opt) => {
    let options = {
        json: 'json',
        csv: 'csv'
    }
    return options[opt]
}
```
@ccali11 I agree with you regarding the switch statement. For now we only have two options so I think an if/else or a switch statement as we have it would be enough.

Reply:
Just submitted my comment above, not sure if you saw it before @hawyar

Reply:
@shanejearley I like this approach it looks great. Do you want to add that in a future PR?

Reply:
Yes I'll copy it down!